packages/manager/apps/dedicated/client/app/billing/confirmTerminate/Service.class.js

```diff
@@ -0,0 +1,13 @@
+export default class Service {
+  constructor(resource) {
+    Object.assign(this, resource);
+  }
+
+  get name() {
+    return this.resource.name;
+  }
+
+  get path() {
+    return this.route.url;
```

Comment:
? where does the `this.route` comes from ?

If someone goes `this.path = a` then you won't have the same value when you `const a = this.path`

Reply:
This comes from API but it's to abstract the way the property is handled by API

Reply:
I understand but it means that the setter and the getter don't work on the same value :)

Reply:
I see your point, I think you are raising a bigger issue because those are not supposed to be updated and we should be able to have readonly properties 
This could be a solution : https://www.typescriptlang.org/docs/handbook/classes.html#readonly-modifier

Reply:
Or you could use a sane alternative to JavaScript which enforces strict immutability 😅